src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1890,3 +1928,44 @@ fn is_primitive(path_str: &str, ns: Namespace) -> Option<(&'static str, Res)> {
 fn primitive_impl(cx: &DocContext<'_>, path_str: &str) -> Option<&'static SmallVec<[DefId; 4]>> {
     Some(PrimitiveType::from_symbol(Symbol::intern(path_str))?.impls(cx.tcx))
 }
+
+fn strip_generic_parameters_from_path(
```

Comment:
nit: this is a little verbose
```suggestion
fn strip_generics(
```

Reply:
What about

```suggestion
fn strip_generics_from_path(
```

?

Reply:
Sounds good to me!